docs/src/examples/lcs/lcs.jl

```diff
@@ -0,0 +1,305 @@
+# straightforward: it only requires annotating the code to expose data
+# dependencies.
+#
+# In our case:
+#
+# - the initialization task writes to the whole `L` array;
+#
+# - filling a tile involves reading `L` for the provided ranges of indices, and
+#   writing to a range of indices shifted by 1;
+#
+# - backtracking reads the whole `L` array.
+#
+# Note that the backtracking task needs to be fetched in order to get the result
+# in a synchronous way and perform an apple-to-apple comparison to the previous
+# implementations.
+
+using DataFlowTasks
+
+function LCS_par(x, y, nx, ny)
+    L = Matrix{Int}(undef, 1+length(x), 1+length(y))
+    DataFlowTasks.@spawn init_lengths!(@W(L), x, y) label="init"
+
+    for (kx, irange) in enumerate(SplitAxis(eachindex(x), nx))
+        for (ky, jrange) in enumerate(SplitAxis(eachindex(y), ny))
+            DataFlowTasks.@spawn begin
+                @R view(L, irange, jrange)
+                @W view(L, irange .+ 1, jrange .+ 1)
+                fill_lengths!(L, x, y, irange, jrange)
+            end label="tile ($kx, $ky)"
+        end
+    end
+
+    bt = DataFlowTasks.@spawn backtrack(@R(L), x, y) label="backtrack"
+    fetch(bt)
+end
+
+# Again, we can check that this implementation produces the correct results, and
+# measure its run-time.
+
+par = LCS_par(x, y, nx, ny)
+@assert seq == par
+t_par = @belapsed LCS_par($x, $y, nx, ny)
+
+# As an added safety measure, let's also check that the task dependency graph
+# looks as expected:
+
+resize!(DataFlowTasks.get_active_taskgraph(), 200)
+GC.gc()
+log_info = DataFlowTasks.@log LCS_par(x, y, nx, ny)
+
+DataFlowTasks.stack_weakdeps_env!()
+using GraphViz
+dag = GraphViz.Graph(log_info)
+DataFlowTasks.savedag("lcs-dag.svg", dag) #src
+
+
+# ## Performance comparison
+#
+# Comparing the performances of these 3 implementations shows that the tiled
+# version already saves some time due to cache effects.
+
+using CairoMakie
+barplot(1:3, [t_seq, t_tiled, t_par],
+        axis = (; title = "Run times [s]",
+                xticks = (1:3, ["sequential", "tiled", "parallel"])))
+
+# The speed-up of the parallel version does not seem ideal, however:
+
+(;
+ nthreads = Threads.nthreads(),
+ speedup  = t_tiled / t_par)
+
+#  Let's try and understand why. The run-time data collected above can help build
+# a profiling plot, which gives some insight about the performances of our
+# parallel version:
+
+plot(log_info, categories=["init", "tile", "backtrack"])
+
+# Here, we see for example that the run time is bounded by the length of the
+# critical path. Adding more threads would not help, and dividing the problem
+# into smaller chunks probably would not help much, either.
+#
+# Indeed, it appears that a non-negligible fraction of the time is spent in the
+# initialization of the array, which is mostly the memory allocation time for
+# the array `L`. This is a limitation of the current implementation, in which
```

Comment:
It seems from the code that the `init` does not take into account the allocation of
the matrix `L`.